Created Android doc for reviewing strings (#248)
flodolo/documentation: src/products/android-l10n/review_Android_strings.md

# Reviewing strings for a new release of Mozilla Android products

Usually, a new release of our Android products means an update to strings. If this is the case, pull requests will be made throughout the release cycle by the mobile team, in order to land these new strings in the [android-l10n](https://github.com/mozilla-l10n/android-l10n) repository in `mozilla-l10n`.

The PRs (also marked as `quarantine`) are reviewed by an l10n-driver - most often by the PM in charge of the project.

Let’s go over some of the steps needed over time in order to review strings correctly for a new release.

## Reviewing the PR

Let's consider Firefox for Android as an example - in fact, these apply to other Android products as well, and they also follow the same train schedule. During the [Android l10n cycle](https://docs.google.com/spreadsheets/d/1hER_w7pOsWSjeBaUqMd_9ClYCCGlwc4jaFYpcBvFE8g/edit#gid=1960046531), the Android team will make pull requests, in order to land new strings for the upcoming release.

Let’s consider a past PR [here](https://github.com/mozilla-l10n/android-l10n/pull/391/files).

> The first file present is `_meta/mozilla-mobile-fenix.json`. Updates to the `main` section are expected and can be ignored.

Reviewer: Why are they expected? What does that file track? ;-)

Author: It looks like some kind of automation for android-l10n: https://github.com/mozilla-l10n-automation-bot/android-l10n/blame/master/_meta/mozilla-mobile-fenix.json
I honestly don't know more, though...
I would say it updates the revision number of main branch each time

Reviewer: Part of writing documentation is learning more around things that are initially not clear ;-) 

That is indeed the hash of a [changeset](https://github.com/mozilla-mobile/fenix/commit/76c0c4ad1425cfd79c25d707921e8185620ad080) from the fenix repository. It's the changeset that is used to generate the strings in that export/PR

Author: Makes total sense :)
Thanks for detailing. I'll update this as well

> The first file present is `_meta/mozilla-mobile-fenix.json`. This file is part of a [changeset](https://github.com/mozilla-mobile/fenix/commit/76c0c4ad1425cfd79c25d707921e8185620ad080) from the Fenix repository, and is used to generate the strings in export PRs. Updates to the `main` section are expected and can be ignored.

Reviewer: This is still not really accurate for someone who's never seen this project (and is the target of these docs).

```suggestion
The first file present is `_meta/mozilla-mobile-fenix.json`. This file is managed by the automation syncing content between code and l10n repositories, and it’s used to track which [changeset](https://github.com/mozilla-mobile/fenix/commit/76c0c4ad1425cfd79c25d707921e8185620ad080) from the `fenix` repository was used to export the strings in this pull requests. `main` is the name of the branch from which strings are extracted, `fenix` is the internal project name of Firefox for Android. Updates to the value of `main` can be ignored, while changes to the name of the branch need to be coordinated with developers and release engineering.
```

Author: Oh ok will update, didn't realize we would need to get into these details

The second file - `values/strings.xml` - is the actual strings file, containing all needed updates.

### Acceptable changes

> The first [string change](https://github.com/mozilla-l10n/android-l10n/pull/391/files#diff-b8d7151f11faa90ad8cfbb96437f96d05602954aa79c1145cb3232f6a5eb6d38R52) adds attributes: `string moz:removedIn="94" name="recently_bookmarked" tools:ignore="UnusedResources"`. You can ignore these kind of changes, as they will not affect localizations. They are a reminder for mobile developers that they will be safely able to delete a string once it has finished riding the Nightly/Beta/Release train, and is no longer needed in an upcoming version. More details [here](https://github.com/mozilla-mobile/fenix/wiki/Removing-strings).

Reviewer: ```suggestion
The first [string change](https://github.com/mozilla-l10n/android-l10n/pull/391/files#diff-b8d7151f11faa90ad8cfbb96437f96d05602954aa79c1145cb3232f6a5eb6d38R52) adds attributes: `string moz:removedIn="94" name="recently_bookmarked" tools:ignore="UnusedResources"`. You can ignore this kind of changes, as they will not affect localizations. They are a reminder for mobile developers that they will be safely able to delete a string once it has finished riding the Nightly/Beta/Release train, and is no longer needed in an upcoming version. More details [here](https://github.com/mozilla-mobile/fenix/wiki/Removing-strings).
```

Changes to comments are irrelevant in terms of string updates, so they’re also OK.

The general review of strings, such as [this one](https://github.com/mozilla-l10n/android-l10n/pull/391/files#diff-b8d7151f11faa90ad8cfbb96437f96d05602954aa79c1145cb3232f6a5eb6d38R662), remains about the same as the usual string review process for other Mozilla products.

Things to look out for:
* Unclear strings and missing localization comments: the best way to identify them is to translate the strings, only having the string and comment as context (not the entire file, or the bug). For example: is the word used both a noun and a verb in English? Is the ID clear enough to give context (e.g. `buttonLabel`)?

> * String changes without new IDs.

Reviewer: Maybe call out what's the ID in a XML file?

> * String changes without new IDs. IDs in XML files are typically located right below the localization comment, and start with `<string name=`, followed by the string ID in quotes - which is then followed by the actual string. So for example [here](https://github.com/mozilla-l10n/android-l10n/pull/391/files#diff-b8d7151f11faa90ad8cfbb96437f96d05602954aa79c1145cb3232f6a5eb6d38R223), the string ID is `browser_menu_customize_home`.

Reviewer: Better to not talk about order, as that's not significant in XML and can change.

```suggestion
* String changes without new IDs. IDs in XML files are stored in the `name` attribute; for example, the ID of [this string](https://github.com/mozilla-l10n/android-l10n/pull/391/files#diff-b8d7151f11faa90ad8cfbb96437f96d05602954aa79c1145cb3232f6a5eb6d38R223) is `browser_menu_customize_home`.
```
